src/v/storage/types.cc

```diff
@@ -60,6 +60,11 @@ std::ostream& operator<<(std::ostream& o, const log_reader_config& cfg) {
     o << ", over_budget:" << cfg.over_budget;
     o << ", strict_max_bytes:" << cfg.strict_max_bytes;
     o << ", skip_batch_cache:" << cfg.skip_batch_cache;
+    o << ", abortable:" << cfg.abort_source.has_value();
+    o << ", aborted:"
+      << (cfg.abort_source.has_value()
```

Comment:
right unclear to me why we'd ever have a config w/out an abort source. doesn't make sense to be optional, no?

Reply:
There are unit/fixture tests that don't wire up the correct lifecycle, that should be fixed of course, but the optionality already existed.